### PR TITLE
Add google consent, DEV-406

### PR DESCRIPTION
**Pull request by iranimij**

### CODE REVIEW

File: Model/Config.php [hunk 1]
```diff
@@ -43,4 +44,9 @@ public function useEuCdn(): bool
     {
         return $this->scopeConfig->isSetFlag(self::XML_PATH_USE_EU_CDN, ScopeInterface::SCOPE_STORE);
     }
+
+    public function googleConsentIsEnabled(): bool
```
Comment (sprankhub) [inline Model/Config.php]: Please rename to `isGoogleConsentModeEnabled` to match the other method names.

File: Model/Config.php [hunk 2]
```diff
@@ -13,6 +13,7 @@ class Config
     public const XML_PATH_COOKIEBOT_ID = 'web/cookiebot/id';
     public const XML_PATH_DATA_CULTURE = 'web/cookiebot/data_culture';
     public const XML_PATH_USE_EU_CDN = 'web/cookiebot/use_eu_cdn';
+    public const XML_PATH_GOOGLE_CONSENT = 'web/cookiebot/google_consent';
```
Comment (sprankhub) [inline Model/Config.php]: Let's rename this setting to `use_google_consent_mode`. The constant name should be names accordigly.

File: view/frontend/templates/script.phtml [hunk 3]
```diff
@@ -2,3 +2,28 @@
 <?php /** @var CustomGento\Cookiebot\ViewModel\Script $viewModel */ ?>
 <?php $viewModel = $block->getData('view_model') ?>
 <?= /* @noEscape */ $viewModel->getScript() ?>
+
+<?php
+if ($viewModel->googleConsentIsEnabled()):
```
Comment (sprankhub) [inline view/frontend/templates/script.phtml]: Please extend the tests to cover this new script.

**Review by sprankhub — CHANGES_REQUESTED** [review]

**Review by sprankhub — APPROVED** [review]

**Comment by Andrii-Antoniuk** [comment]
Thank you for your work! ❤️

**Pull request merged**